### PR TITLE
[TE] frontend - Alert page bug fixes

### PR DESCRIPTION
**Pull request by newsummit**
Handling 3 fixes for Alert Page:
- adding typeahead search in "Dimensions" filter (sometimes we have 200+ options to scroll)
- anomaly table filter by resolution now working properly + added clarity on number of anomalies being displayed, both in section title and index row
- handled case of anomaly index in each row being un-readable (spacing issues)

Also:
- Alert page "resolution options" was displaying duplicates - deduped the array
- Correction to the error state for "we were not able to load all anomalies"

screen shot 2018-10-12 at 5 28 29 pm

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/controller.js [hunk 1]
```diff
@@ -330,28 +335,31 @@ export default Controller.extend({
...
-          anomalies = anomalies.filter(data => targetResolution === data.anomalyFeedback);
+          newAnomalies = newAnomalies.filter(data => targetResolution === data.anomalyFeedback);
         }
+        // Let page know whether anomalies viewed are filtered or not
+        set(this, 'isAnomalyListFiltered', anomalyData.length !== newAnomalies.length);
         // Add an index number to each row
-        anomalies.forEach((anomaly, index) => {
-          anomaly.index = index + 1;
+        newAnomalies.forEach((anomaly, index) => {
+          set(anomaly, 'index', index + 1);
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/controller.js]: set(anomaly, 'index', ++index);

File: thirdeye/thirdeye-frontend/app/styles/components/te-anomaly-table.scss [hunk 2]
```diff
@@ -45,12 +53,16 @@
...
     color: app-shade(black, 6);
-    padding-left: 20px;
+    padding-left: 15px;
     line-height: 45px;
 
     &--fixed {
       width: 215px;
     }
+
+    &--lefter {
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/styles/components/te-anomaly-table.scss]: nip: &--left only is simple and clean name

File: thirdeye/thirdeye-frontend/app/styles/components/te-anomaly-table.scss [hunk 3]
```diff
@@ -71,9 +83,13 @@
 
   &__list {
     list-style: none;
-    margin-left: -39px;
+    margin-left: -37px;
     margin-bottom: 0;
     max-width: 280px;
+
+    &--lefter {
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/styles/components/te-anomaly-table.scss]: nip: &--left only is simple and clean name

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/controller.js [hunk 4]
```diff
@@ -186,6 +189,7 @@ export default Controller.extend({
     'currentPage',
     'loadedWoWData',
     'selectedSortMode',
+    'selectedResolution',
     function() {
       let anomalies = this.get('filteredAnomalies');
       const { pageSize, currentPage, selectedSortMode } = this.getProperties('pageSize', 'currentPage', 'selectedSortMode');
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/controller.js]: `getProperties(this,`  here too to be consistent

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/controller.js (hunk 1 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/controller.js]: thanks

**Pull request merged**